### PR TITLE
Added possibility to compress in CROSS JOIN

### PR DESCRIPTION
**Pull request by p1rattttt**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added possibility to compress in CROSS JOIN

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 4ac32668bd7053d3c094020e75bd1ae5e7187fe5 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60459/4ac32668bd7053d3c094020e75bd1ae5e7187fe5/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>❌ failure</td></tr>
<tbody>
</table>

**Comment by alexey-milovidov** [comment]
@p1rattttt Add some tests to prove that it works.

The first test I tried didn't show the effect:
```
milovidov-desktop :) SET cross_join_compress_blocks = 1

SET cross_join_compress_blocks = 1

Query id: 8c89aa90-b7ab-4007-badd-a9d5cf49f56e

Ok.

0 rows in set. Elapsed: 0.001 sec. 

milovidov-desktop :) SELECT count() FROM hits, hits

... killed
```

**Comment by alexey-milovidov** [comment]
What we actually want is the threshold on data size in memory to start compressing.
So we can enable it by default with some reasonable threshold.

**Comment by p1rattttt** [comment]
> The first test I tried didn't show the effect:

It is because compression of 'hits' table change nothing, better to check it on 'visits'

> Add some tests to prove that it works.
I'll add tests soon

File: src/Core/SettingsChangesHistory.h [hunk 1]
```diff
@@ -88,6 +88,8 @@ static std::map<ClickHouseVersion, SettingsChangesHistory::SettingsChanges> sett
     {"24.5", {{"allow_deprecated_functions", true, false, "Allow usage of deprecated functions"},
               {"allow_experimental_join_condition", false, false, "Support join with inequal conditions which involve columns from both left and right table. e.g. t1.y < t2.y."},
               {"output_format_parquet_use_custom_encoder", false, true, "Enable custom Parquet encoder."},
+              {"cross_join_min_rows_to_compress", 0, 10000000, "A new setting."},
+              {"cross_join_min_bytes_to_compress", 0, 1_GiB, "A new setting."},
```
Comment (Avogar) [inline src/Core/SettingsChangesHistory.h:92]: Please, improve the description. `A new setting.` is not good, it doesn't tell why these settings were added and what they control

**Pull request merged**